README.md

```diff
@@ -1 +1,206 @@
+
+Een authorization_code kan je bekomen door de gebruiker naar de OAuth2 Authorize applicatie te redirecten.
+
+
+#### 2) access_token bekomen
+
+Nadat de gebruiker is ingelogd en zijn goedkeuring(consent) heeft gegeven zal de gebruiker naar de redirect_uri geredirect worden die je hebt ingegeven in de details van uw applicatie op de API store;
+
+Deze redirect_uri bevat een *code* querystring parameter als de authenticatie succesvol is. Bijvoorbeeld:
+https://domain.com/callback?code=123
+
+*Request:*
+
+curl -X POST -d "client_id={{CLIENT_ID}}&client_secret={{CLIENT_SECRET}}&code={{CODE_FROM_PREV_STEP}}&grant_type=authorization_code" https://api-gw-p.antwerpen.be/org/service/v1/oauth2/token
+
+*Response:*
+{"refresh_token":"66f0c43c27a94ad4aa2d7574cf7b4465","token_type":"bearer","access_token":"a2824fb10b2a44b2b6f1a4aba382630a","expires_in":7200}
+
+#### 3) API oproepen
+curl -v -H "Authorization: Bearer {{ACCESS_TOKEN_FROM_PREVIOUS_STEP}}" 'https://api-gw-p.antwerpen.be/org/service/v1/resource'
+
+
+### OAuth2 profiel: implicit
+
+Hoewel het "implicit" OAuth2 profiel beschikbaar is op de API gateway raden we aan om deze niet te gebruiken.
+
+Voornamelijk omwille van onderstaande redenen:
++ Access_tokens die zijn aangemaakt met de implicit/token flow zijn slechts 2 uur gelding en kunnen niet verlengd worden.
++ Aangezien de client_secret niet gebruikt wordt om een access_token aan te vragen is dit OAuth2 profiel vatbaarder voor security issues.
+
+## Uitloggen
+
+## Het uitloggen van een gebruiker initiëren vanuit uw applicatie
+
+De OAuth2 authorizatie server bevat ook de mogelijkheid om gebruikers centraal te laten uitloggen.
+
+Om dit te bereiken moet je de gebruiker redirecten naar de "/logout/redirect/encrypted" endpoint van de OAuth2 authorizatie server met onderstaande querystring parameters.
+
+| Parameter | Verplicht | Omschrijving |
+| :---         |     :---:      |  :---   |
+| client_id     | true       | De client_id die je kan terugvinden bij uw applicatie in de API store.
+| service     | true       | De service waarmee de gebruiker zich geauthenticeerd heeft. Dit kan 'astad.aprofiel.v1' of 'astad.mprofiel.v1' zijn.      |
+| data     | true       | Een object met extra data ivm het uitloggen. Deze data is geencrypteerd volgens sha1 op basis van de client_secret.      |
+| data.user_id     | true       | De id van de gebruiker die wenst uit te loggen      |
+| data.access_token     | true       | Het access_token van de gebruiker die wenst uit te loggen      |
+| data.redirect_uri     | true       | De url op uw applicatie naar waar de gebruiker geredirect moet worden nadat hij uitgelogd is. Het domein van deze redirect_uri moet overeenkomen met het domein van de redirect_uri(callback) die je in de api-store hebt ingegeven voor het aanlogproces |
+
+
+Naast het centraal uitloggen van de gebruiker is het ook de bedoeling om de sessie in uw applicatie te termineren.
```

Comment:
Nadat de gebruiker centraal is uitgelogd moet je ook nog de gebruiker sessie in jouw applicatie beëindigen.